happiness-backend/api/dao/happiness_dao.py

```diff
@@ -64,7 +66,74 @@ def get_happiness_by_group_count(user_ids, page, n):
         Happiness.timestamp.desc()) \
         .paginate(page=page, per_page=n, error_out=False)
 
+def get_happiness_by_filter(user_id, page, per_page, start, end, low, high, query):
+    """
+    Filters according to the provided arguments. Checks to see what filters to apply. Will not apply the following
+    filters with the following arguments
+     low/high = 11
+     start/end = None
+     text = "".
+    """
+    if start is not None and low != 11 and query != "":
+        return db.paginate(
+            select=
+            (select(Happiness).where(Happiness.timestamp.between(start, end),
+                                     Happiness.user_id == user_id).where(
+                    Happiness.value >= low, Happiness.value <= high,
+                    Happiness.user_id ==
+                    user_id).where(Happiness.comment.like(f"%{query}%")).order_by(Happiness.timestamp.asc())),
```

Comment:
Why are you chaining together `.where` calls? Couldn't you apply all the filters in a single `where`? [Like this](https://stackoverflow.com/a/72828047)

Reply:
‼️

Reply:
lol i didn't know that exists indeed this would be better